.github/workflows/test_installer.yml

```diff
@@ -0,0 +1,138 @@
+        with:
+          verifier-release: v1.0.0
+      - name: Run verifier help
+        run: slsa-verifier --help
+        shell: bash
+  installer_job_v101:
+    runs-on: ubuntu-latest
+    name: Install v1.0.1
+    steps:
+      - name: Install verifier v1.0.1
+        uses: slsa-framework/slsa-verifier/actions/install@main
+        with:
+          verifier-release: v1.0.1
+      - name: Run verifier help
+        run: slsa-verifier --help
+        shell: bash
+  installer_job_v102:
+    runs-on: ubuntu-latest
+    name: Install v1.0.2
+    steps:
+      - name: Install verifier v1.0.2
+        uses: slsa-framework/slsa-verifier/actions/install@main
+        with:
+          verifier-release: v1.0.2
+      - name: Run verifier help
+        run: slsa-verifier --help
+        shell: bash
+  installer_job_v110:
+    runs-on: ubuntu-latest
+    name: Install v1.1.0
+    steps:
+      - name: Install verifier v1.1.0
+        uses: slsa-framework/slsa-verifier/actions/install@main
+        with:
+          verifier-release: v1.1.0
+      - name: Run verifier help
+        run: slsa-verifier --help
+        shell: bash
+  installer_job_v111:
+    runs-on: ubuntu-latest
+    name: Install v1.1.1
+    steps:
+      - name: Install verifier v1.1.1
+        uses: slsa-framework/slsa-verifier/actions/install@main
+        with:
+          verifier-release: v1.1.1
+      - name: Run verifier help
+        run: slsa-verifier --help
+        shell: bash
+  installer_job_v120:
+    runs-on: ubuntu-latest
+    name: Install v1.2.0
+    steps:
+      - name: Install verifier v1.2.0
+        uses: slsa-framework/slsa-verifier/actions/install@main
+        with:
+          verifier-release: v1.2.0
+      - name: Run verifier help
+        run: slsa-verifier --help
+        shell: bash
+  installer_job_v130:
+    runs-on: ubuntu-latest
+    name: Install v1.3.0
+    steps:
+      - name: Install verifier v1.3.0
+        uses: slsa-framework/slsa-verifier/actions/install@main
+        with:
+          verifier-release: v1.3.0
+      - name: Run verifier help
+        run: slsa-verifier --help
+        shell: bash
+  installer_job_main:
+    runs-on: ubuntu-latest
+    name: Install from main
+    steps:
+      - name: Install go
+        uses: actions/setup-go@v3
+        with:
+          go-version: 1.18
+          check-latest: true
+      - name: Install verifier at main
+        uses: slsa-framework/slsa-verifier/actions/install@main
+        with:
+          verifier-release: main
+      - name: Run verifier help
+        run: slsa-verifier --help
+        shell: bash
+  installer_job_custom_path:
+    runs-on: ubuntu-latest
+    name: Install in custom directory
+    steps:
+      - name: Install verifier v1.3.0
+        uses: slsa-framework/slsa-verifier/actions/install@main
+        with:
+          install-dir: /tmp/.slsa-verifier
+      - name: Run verifier help
+        run: slsa-verifier --help
+        shell: bash
+  installer_job_sudo_enabled:
```

Comment:
I can't think of a way to test failure due to hash mismatch or provenance verification failure on the fetched binary. If there's a way we can do that., that'd be cool. Bu don't waste time on it :)

Reply:
I added a couple of special version tags that will force checksum and provenance errors and wrote tests that use them. I don't love the solution, but it works. WDYT?